File: Daily challenge/Largest Positive Integer That Exists With Its Negative.js

var findMaxK = function(nums) 
{
    nums.sort((a, b) => b - a);
    for (let n of nums) 
    {
        if (n > 0 && nums.includes(-n)) 
        {
            return n;
        }
    }
    return -1;    
};

File: Daily challenge/Largest Positive Integer That Exists With Its Negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) 
    {
        sort(nums.begin(), nums.end());
        reverse(nums.begin(), nums.end());
        for (int i = 0; i < nums.size(); i++)
        {
            if (nums[i] > 0 && find(nums.begin(), nums.end(), -nums[i]) != nums.end()) 
            {
                return nums[i];
            }
        }
        return -1;
    }
};